pyeapi/api/vrfs.py

```diff
@@ -0,0 +1,358 @@
+        if match:
+            value = match.group('value')
+        else:
+            value = match
+        return dict(rd=value)
+
+    def _parse_description(self, config):
+        """ _parse_description scans the provided configuration block and
+        extracts the vrf description value. The return dict is intended to
+        be merged into the response dict.
+
+        Args:
+            config (str): The vrf configuration block from the nodes
+                running configuration
+
+        Returns:
+            dict: resource dict attribute
+        """
+        value = DESCRIPTION_RE.search(config).group('value')
+        return dict(description=value)
+
+    def getall(self):
+        """Returns a dict object of all VRFs in the running-config
+
+        Returns:
+            A dict object of VRF attributes
+
+        """
+        vrfs_re = re.compile(r'(?<=^vrf definition\s)(\w+)', re.M)
+
+        response = dict()
+        for vrf in vrfs_re.findall(self.config):
+            response[vrf] = self.get(vrf)
+        return response
+
+    def create(self, vrf_name):
+        """ Creates a new VRF resource
+
+        Args:
+            vrf_name (str): The VRF name to create
+
+        Returns:
+            True if create was successful otherwise False
+        """
+        command = 'vrf definition %s' % vrf_name
+        return self.configure(command)
+
+    def delete(self, vrf_name):
+        """ Deletes a VRF from the running configuration
+
+        Args:
+            vrf_name (str): The VRF name to delete
+
+        Returns:
+            True if the operation was successful otherwise False
+        """
+        command = 'no vrf definition %s' % vrf_name
+        return self.configure(command)
+
+    def default(self, vrf_name):
+        """ Defaults the VRF configuration for given name
+
+        Args:
+            vrf_name (str): The VRF name to default
+
+        Returns:
+            True if the operation was successful otherwise False
+        """
+        command = 'default vrf definition %s' % vrf_name
+        return self.configure(command)
+
+    def configure_vrf(self, vrf_name, commands):
+        """ Configures the specified VRF using commands
+
+        Args:
+            vrf_name (str): The VRF name to configure
+            commands: The list of commands to configure
+
+        Returns:
+            True if the commands completed successfully
+        """
+        commands = make_iterable(commands)
+        commands.insert(0, 'vrf definition %s' % vrf_name)
+        return self.configure(commands)
+
+    def set_rd(self, vrf_name, rd):
+        """ Configures the VRF rd
+
+        EosVersion:
+            4.xx.xx
+
+        Args:
+            vrf_name (str): The VRF name to set rd for
+            rd (str): The value to configure the vrf rd
+
+        Returns:
+            True if the operation was successful otherwise False
+        """
+        if not isrd(rd):
+            return False
```

Comment:
So is the recommendation here to remove the rd check in its entirety?

The error returned from EOS is 'Ambiguous command' for an incorrect rd format and that error is stored in the connection object instead of raising the exception.